includes/formatting.php

```diff
@@ -122,3 +175,47 @@ function strip_wrapping_div( $content ) {
 	return trim( preg_replace( $regex, '$1', trim( $content ) ) );
 }
 add_filter( 'airstory_before_insert_content', __NAMESPACE__ . '\strip_wrapping_div' );
+
+/**
+ * When inserting new attachments, set the post_author to match the current user.
+ *
+ * @param array $post The attachment's post object.
+ * @return The $post array, with post_author set to match the author of the attachment's parent.
+ */
+function set_attachment_author( $post ) {
+	if ( ! empty( $post['post_author'] ) ) {
+		return $post;
+	}
+
+	$parent_post = get_post( $post['post_parent'] );
+
+	if ( ! $parent_post ) {
+		return $post;
+	}
+
+	$post['post_author'] = (int) $parent_post->post_author;
+
+	return $post;
+}
+
+/**
+ * Get the ID of an attachment based on its (post-sideload) URL.
+ *
+ * This is a bit of a workaround, as WordPress doesn't expose a native method of retrieving an
+ * attachment's ID based on its URL. Instead, we'll use a quick database query.
+ *
+ * @link https://pippinsplugins.com/retrieve-attachment-id-from-image-url/
+ *
+ * @global $wpdb
+ *
+ * @param string $attachment_url The attachment URL.
+ * @return int Either an attachment ID or 0 if no matching attachment was found.
+ */
+function get_attachment_id_by_url( $attachment_url ) {
+	global $wpdb;
+
+	return (int) $wpdb->get_var( $wpdb->prepare(
+		"SELECT ID FROM $wpdb->posts WHERE post_type = 'attachment' AND guid = %s LIMIT 1;",
```

Comment:
I'm sure you know, but this is kinda flaky, especially since the GUIDs can be changed or modified by other things.